parcel/ParcelTendencies.jl

```diff
@@ -106,6 +107,39 @@ function immersion_freezing(params::P3_het, PSD, state)
     return max(FT(0), (Nᵢ_het - Nᵢ) / const_dt)
 end
 
+function immersion_freezing(params::Frostenberg_random, PSD, state)
+    FT = eltype(state)
+    (; ip, drawing_interval) = params
+    (; T, Nₗ, Nᵢ, t) = state
+    if mod(t, drawing_interval) == 0
+        μ = CMI_het.INP_concentration_mean(T)
+        INPC = exp(rand(Normal(μ, ip.σ)))
+    else
+        INPC = 0
+    end
+    return min(Nₗ, max(FT(0), INPC - Nᵢ))
```

Comment:
All the `immersion_freezing` functions should be returning the freezing rate. Thats the expectation of the ODE solver  - we can't overwrite the sate, we can only define the tendencies for the solver. So you should divide what you are returning here by `dt` (It shouldn't change the results much since I think by default you were running with `dt=1s`)